Worker properties send dicts

bruyeret opened the pull request:
## Add possibility to use dict as property values

Remove the restriction on NaN values for properties (too tedious to check and should be done by the client, not the server)

Properties can have "sub-properties"
The architecture enables the use of dict of dict without depth limitation
Keys in the dict can be any valid mongoDB field (see [mongodb field names restrictions](https://www.mongodb.com/docs/v3.2/reference/limits/#Restrictions-on-Field-Names))

For the annotation client:
- addAnnotationPropertyValues is retrocompatible and accept more types for the "value" argument
- getPropertyHistogram is also retrocompatible, the propertyPath is more complex than before (see docs)

The server Docker image needs to be rebuilt (`docker-compose build girder`).

### Before this PR is merged

To be able to send dict property values using the ImageAnalysis project docker images:
- if you use `addAnnotationPropertyValues`, it should be fine with the cached version of the annotation client
- if you use `add_annotation_property_values`, you should use the new version of the annotation client by adding these two lines right after `WORKDIR /UPennContrast`:
```
RUN git fetch --all
RUN git switch worker-tuples
```

### After this PR is merged

To use the new annotation client, rebuild the docker images using `--no-cache` flag so that it clones the new version of the repo instead of using the cached one.


Close #451 
Close #470

arjunrajlab commented:
@bruyeret Looks great! Minor issue:

I made a property with a few dictionary elements:

<img width="597" alt="image" src="https://github.com/Kitware/UPennContrast/assets/13998824/981b2a67-3409-40db-b1d7-c88aa3583a7d">

The problem is that in the annotation list, it shows the "parent" property as a column. Like here, it shows cell "Blob Blob metrics" as a column, but there is no actual property for that "parent".

<img width="615" alt="image" src="https://github.com/Kitware/UPennContrast/assets/13998824/32d68bca-b277-4aa5-b28b-dbf6bb4b6776">

bruyeret commented:
I added some checks on properties and removed the "show property when created" feature as it is impossible to know which paths are valid before computing the property
What do you think @arjunrajlab? Ready to merge?

arjunrajlab commented:
@bruyeret Sorry for the delay, just tested again, looks great! Ready to merge!

luciemac commented:
LGTM
